sdks/v4-sdk/src/entities/position.ts

```diff
@@ -52,59 +52,262 @@ export class Position {
    * Returns the price of token0 at the lower tick
    */
   public get token0PriceLower(): Price<Currency, Currency> {
-    // TODO: Currency or wrapped token here?
-    throw new Error('Not implemented')
+    return tickToPrice(this.pool.currency0, this.pool.currency1, this.tickLower)
   }
 
   /**
    * Returns the price of token0 at the upper tick
    */
   public get token0PriceUpper(): Price<Currency, Currency> {
-    throw new Error('Not implemented')
+    return tickToPrice(this.pool.currency0, this.pool.currency1, this.tickUpper)
   }
 
   /**
    * Returns the amount of token0 that this position's liquidity could be burned for at the current pool price
    */
   public get amount0(): CurrencyAmount<Currency> {
-    throw new Error('Not implemented')
+    if (this._token0Amount === null) {
```

Comment:
`if (!this._token0Amount)` will include the case it's `undefined`

Reply:
oh got it cool! updated. out of curiousity, could it ever be set to undefined instead of null ?

Reply:
Oh no, you're right.. I see it being initialized to null.